### PR TITLE
Feature/Add category groups to report vulnerability

### PR DESCRIPTION
**Pull request by oussamaessaji**

### CODE REVIEW

File: src/ostorlab/agent/kb/kb.py [hunk 1]
```diff
@@ -47,6 +48,11 @@ def __getattr__(cls, item: str) -> Entry:
             encoding="utf-8"
         ) as r:
             meta = json.loads(f.read())
+            category_groups = []
+            for category_key, categories_list in meta.get("categories", {}).items():
+                category_group = {"key": category_key, "categories": categories_list}
+                category_groups.append(category_group)
```
Comment (deadly-panda) [inline src/ostorlab/agent/kb/kb.py:54]: Could go for list comprehension no ?
categories = meta.get("categories", {})
category_groups = [{'key': k, "categories": v} for k, v in categories.items()]

Comment (benyissa) [reply src/ostorlab/agent/kb/kb.py:54]: +1.

File: tests/agent/mixins/agent_report_vulnerability_mixin_test.py [hunk 2]
```diff
@@ -151,3 +151,39 @@ def testSendVulnerabilityMessage_whenKbEntryIsValidAndRiskIsCritical_emitVulnera
...
+def testSendVulnerabilityMessage_whenKbEntryIsValidWithCategories_emitVulnerabilityMessage(
+    agent_mock: list[object],
+) -> None:
+    """Prepares & emit vulnerability message with all the information from the Knowledge base."""
+    agent_definition = agent_definitions.AgentDefinition(
+        name="some_name", out_selectors=["v3.report.vulnerability"]
+    )
+    agent_settings = runtime_definitions.AgentSettings(key="some_key")
+    report_vul_mixin = TestAgent(agent_definition, agent_settings)
+    report_vul_mixin.report_vulnerability(
```
Comment (benyissa) [inline tests/agent/mixins/agent_report_vulnerability_mixin_test.py:165]:
```suggestion

    report_vul_mixin.report_vulnerability(
```

File: src/ostorlab/agent/kb/kb.py (hunk 1 above)
Comment (oussamaessaji) [reply src/ostorlab/agent/kb/kb.py]: I can do it!

**Review by benyissa — DISMISSED** [review]

File: tests/agent/mixins/agent_report_vulnerability_mixin_test.py [hunk 3]
```diff
@@ -151,3 +151,40 @@ def testSendVulnerabilityMessage_whenKbEntryIsValidAndRiskIsCritical_emitVulnera
     assert agent_mock[0].data["references"] == [
         {"title": "Virustotal", "url": "https://www.virustotal.com/"}
     ]
+
+
+def testSendVulnerabilityMessage_whenKbEntryIsValidWithCategories_emitVulnerabilityMessage(
+    agent_mock: list[object],
+) -> None:
+    """Prepares & emit vulnerability message with all the information from the Knowledge base."""
```
Comment (benyissa) [inline tests/agent/mixins/agent_report_vulnerability_mixin_test.py:159]:
```suggestion
    """Prepares & emit vulnerability message with all the information from the knowledge base."""
```

**Review by benyissa — APPROVED** [review]

File: src/ostorlab/agent/kb/kb.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import json
 import pathlib
 
-from typing import Dict
+from typing import Dict, Optional, Union
```
Comment (3asm) [inline src/ostorlab/agent/kb/kb.py:6]: Please use the modern types instead. `dict`, | and | None.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/agent/kb/kb.py (hunk 4 above)
Comment (oussamaessaji) [reply src/ostorlab/agent/kb/kb.py:6]: They are not supported in python3.9

**Review by 3asm — APPROVED** [review]

**Pull request merged**